### DISCUSSION TITLE
Link to the internal page of the CMS

### DISCUSSION THREAD

**Discussion by esisl** [pimcore/pimcore#8994] [Q&A / Support] [answered]
How to make a link to the internal page of the CMS, to an object in Date Objects?
I need to send a link by mail, so that an authorized user can follow this link to a tab with an object for editing.

Is it possible in the CMS forcibly, with the php code, to open a tab with an object for editing?

**Comment by aarongerig** [comment, marked as answer]
Hi @esisl

Yes, this feature is called "deeplink". Here's more on that: https://pimcore.com/docs/pimcore/current/Development_Documentation/Extending_Pimcore/Deeplinks_into_Admin_Interface.html

To generate that route, you can use following schema:
```php
/**
 * Generates a link like: https://acme.com/admin/login/deeplink?{type}_{ID}_{subtype}
 */
public function generateDeeplink(string $type, int $id, string $subtype)
{
    return sprintf(
        '%s?%s_%s_%s',
        $this->router->generate('pimcore_admin_login_deeplink'),
        $type,
        $id,
        $subtype
    );
}

```

**Comment by esisl** [comment]
O! Thank you!
